compiler/rustc_hir_analysis/src/structured_errors/wrong_number_of_generic_args.rs

```diff
@@ -129,6 +129,46 @@ impl<'a, 'tcx> WrongNumberOfGenericArgs<'a, 'tcx> {
         if self.missing_lifetimes() { "lifetime" } else { "generic" }
     }
 
+    fn is_trait(&self) -> bool {
+        self.tcx.is_trait(self.def_id)
+    }
+
+    /// Returns true if the generic type is a trait
+    /// and is being referred to from one of its trait impls
+    fn is_in_trait_impl(&self) -> bool {
+        self.is_trait() && {
```

Comment:
what's up with the extra block here? I would just use an if statement.

Reply:
Guess I overthought this one a bit.

Fixed.